lib/propmangle.js

```diff
@@ -159,7 +162,7 @@ function mangle_properties(ast, options) {
         if (options.only_cache) {
             return cache.props.has(name);
         }
-        if (/^[0-9.]+$/.test(name)) return false;
+        if (/^-?[0-9]+(\.[0-9]+)?(e[+-][0-9]+)?$/.test(name)) return false;
```

Comment:
We don't actually want to exclude strings like `"0x25"` - we only need to avoid `"37"`.